### DISCUSSION TITLE
Updated Standard File Naming Convention

### DISCUSSION THREAD

**Discussion by CodyBerenson** [CodyBerenson/PGMA-Modernized#130] [General]
The standard file naming convention has been updated:

Name your files as: (Studio; IAFD Studio) -  Title (Year) - [Cast List] - Stack Type Stack Number

Notes:  
•	Please see usage guide to understand optional vs. mandatory   
•	Cast List must be enclosed in square brackets and each name separated by Commas. If a cast list is detected the agents will use it instead of the cast found on the agents website.  This resolves issues with inconsistent/non-existent actor tagging with certain index sites (e.g.,Best Gay Exclusive and Queer Click)
•	Stack type = part, cd, dvd, disc,disk etc
•	Stack no... 1 to 8

**Comment by jfreiman** [comment]
This looks to be helpful when adding difficult or new videos to Collections.  My question is, will the 'old' naming convention continue to work as before? or will our existing videos names need to be updated/changed in order for metadata to be refreshed or re-matched?

**Comment by JPH71** [comment]
You do not need to change any of the old titles. This functionality was
introduced to help match films in IAFD as sometimes IAFD uses a completely
different Studio name to the Studio in an agent like GEVI.
In cases where your have next door studios in GEVI and nextdoorbuddies.com
in IAFD, you can get away with having the 2 studios mentioned... for
example:
(NextDoorBuddies;Next Door) - Best Buds (2021)
could be just named:
(Next Door) - Best Buds (2021) as Next Door is in both studio names...

Studio name comparison ignores spaces...

**Comment by jfreiman** [comment]
Ok, I'm just getting around to changing how I add cast lists to Titles not matching in IAFD and/or other databases.

For me....  it's not working.  here are two examples and no Cast is present after loading into Plex.

The Library is set up with only Scenes and I have the following agents under Gay Adult Scenes selected:
- Gay Adult Scenes
- Internet Adult Film Database
- WayBig (IAFD)
- GEVIScenes (IAFT)
- Local Media Assets (Movies)

These are the filenames:
(ragingstallion.com) - Jack of All Trades (2024) - [Johnny Ford, Greg Dixxon].mp4
(ragingstallion.com) - Jack of All Trades (2024) - [Tryp Bates, Drew Valentino].mp4

![image](https://github.com/CodyBerenson/PGMA-Modernized/assets/10014964/1492e9e7-baf6-42f7-8bc3-10d3523dcc61)

I have tried moving them out of my scanned folders, deleted the entries in Plex, cleaned the database and then re-added the files back and have the same issue.

Do I need to add another agent added?  Other titles are correctly added by IAFD - ie. the files/videos which do have entries in IAFD.

**Comment by JPH71** [comment]
Give me 20 minutes to look at this and I will let you know...

**Comment by JPH71** [comment]
I can't find either of the scenes on waybig so no go there...

For GeviScenes, you have them named wrong they should be named as follows

(Raging Stallion Studios) - Jack of All Trades - {191386N} (2023) - [Tryp
Bates,Drew Valentino

Remember the naming rules for GEVIScenes are different from the other
agents, as there s no search functionality for GEVIScenes it needs the
scene number in the address to reference its page. That is why there is
{191386N} in the title, The N after the number is to deal with image
rotation. In this case none. L would rotate Left and R would rotate Right.

**Reply by jfreiman** [reply]
> Remember the naming rules for GEVIScenes are different from the other
agents, as there s no search functionality for GEVIScenes it needs the
scene number in the address to reference its page. That is why there is
{191386N} in the title, The N after the number is to deal with image
rotation. In this case none. L would rotate Left and R would rotate Right.

OMG - You're the BEST!

In my example/attempt I gave above, I wasn't really trying to match the GEVI Scene(s), I was just using a similar naming struture to match the rest of my collection.

I was really stuck with getting the Cast List to pull the Cast Artwork.  

The examples you gave me work and I'll play around with trying to convert my existing "unmatchable" (eric videos for example) with this method vs adding them into the mp4 metadata as "artist" - which adds the cast to PMS, but with no artwork which 'breaks' other videos those actors are in (erases their photo).

Thanks again for your HELP! 

-John

**Comment by jfreiman** [comment]
Well, I tried using your example above and apply it to one of my Eric Videos - not in GEVI or AFID.

I am expecting that PGMA will not find the video in the database and "fall back" to the data I'm providing in the filename - Studio, Title, Date, and find and add the Cast List from IAFD's images - where available.

(ERIC VIDEOS) - Twink Addicted To Juice (2023) - [Alex,Gabriel,Justin Jett].mp4
** Obviously, I don't expect this example to properly match "Alex' and "Gabriel" but Justin Jett is in IAFD with a photo and that's what I'd expect to appear in Plex....

When added to PMS with this filename, there is no Cast List provided. 

Is this the way the feature is supposed to work?

**Reply by CodyBerenson** [reply]
Hi @jfreiman,

I think you misunderstand what the agents and the - [Cast List] will do.  Sorry, The - [Cast List] only works when the PGMA agent is able to make a match.  In fact, if our agents can't make a match, they don't "fall back" and populate any metadata, regardless of what's in the filename or the file properties.  Any metadata that you are adding to the files properties is populated in Plex by the "Local Media Assets" standard Plex agent.  

FYI, In the "unmatchable" use case you provide (e.g., many Eric Video titles), an alternate to populating a file's properties you may want to consider using the GAnfoMoviesImporter agent to populate full metadata, including cast and posterart.  This agent doesn't try to match from any indexing site like GEVI.  It uses XML data that you have populated in a .nfo file that is paired alongside with your movie and poster image.   Instructions for how to do this, along with examples, begin on page 28 of the usage guide.  I've queried the community before....some folks like using notepad to create the .nfo files, while others use one of multiple desktop apps.  I mostly use TinyMediaManager.  

Once upon a time, GAnfoMoviesImporter retrieved cast photos from IAFD (which is ideally what you're looking for).  That's been broken for a while because of IAFD's cloudflare, and its time for me to needle Jason with a reminder to look into fixing it.

Hope this clears things up.  Cheers.

**Reply by jfreiman** [reply]
That does help.  I read, and re-read the instructional pdf and it made me think that matching wasn't necessary.
I'll look into creating XML files in the future.
Right now I like the simplicity of using Foobar2000 - it's a nearly 1-stop shop me to name, re-name, and standardize on both the Studio and Distributor names, etc.  I do have to use a second program to add the video description, but it's all very straightforward....

Switching to XML would be nice since it will (potentially) pull the Cast images - that's pretty sweet.

Thanks for your help.

**Reply by CodyBerenson** [reply]
Terrific!  I've reminded @JPH71 to look into the IAFD portion of GAnfoMoviesImporter when he can find some time.  Cheers!
